// @ngu/carousel/esm5/lib/ngu-item/ngu-item.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
import { Component, HostBinding } from '@angular/core';
import * as ɵngcc0 from '@angular/core';

var _c0 = ["*"];
var NguItemComponent = /** @class */ (function () {
    function NguItemComponent() {
        this.classes = true;
    }
    NguItemComponent.propDecorators = {
        classes: [{ type: HostBinding, args: ['class.item',] }]
    };
NguItemComponent.ɵfac = function NguItemComponent_Factory(t) { return new (t || NguItemComponent)(); };
NguItemComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: NguItemComponent, selectors: [["ngu-item"]], hostVars: 2, hostBindings: function NguItemComponent_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵclassProp("item", ctx.classes);
    } }, ngContentSelectors: _c0, decls: 1, vars: 0, template: function NguItemComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵprojection(0);
    } }, styles: [""] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NguItemComponent, [{
        type: Component,
        args: [{
                // tslint:disable-next-line:component-selector
                selector: 'ngu-item',
                template: "<ng-content></ng-content>\n",
                styles: [""]
            }]
    }], function () { return []; }, { classes: [{
            type: HostBinding,
            args: ['class.item']
        }] }); })();
    return NguItemComponent;
}());
export { NguItemComponent };
if (false) {
    /** @type {?} */
    NguItemComponent.prototype.classes;
}
